src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -58,36 +69,59 @@ zocl_fpga_mgr_load(struct drm_zocl_dev *zdev, const char *data, int size, u32 fl
 	return err;
 }
 
+/*
+ * Load partial bitstream to PR platform
+ *
+ * @param       zdev:   zocl device structure
+ * @param       buffer: xclbin data buffer pointer
+ * @param       length: xclbin data buffer size
+ * @param       domain: Target domain structure
+ *
+ * @return      0 on success, Error code on failure.
+ */
 static int
-zocl_load_partial(struct drm_zocl_dev *zdev, const char *buffer, int length)
+zocl_load_partial(struct drm_zocl_dev *zdev, const char *buffer, int length,
+		  struct drm_zocl_domain *domain)
 {
 	int err;
 	void __iomem *map = NULL;
 
-	if (!zdev->pr_isolation_addr) {
+	if (!domain->pr_isolation_addr) {
 		DRM_ERROR("PR isolation address is not set");
 		return -ENODEV;
 	}
 
-	map = ioremap(zdev->pr_isolation_addr, PR_ISO_SIZE);
+	map = ioremap(domain->pr_isolation_addr, PR_ISO_SIZE);
 	if (IS_ERR_OR_NULL(map)) {
 		DRM_ERROR("ioremap PR isolation address 0x%llx failed",
-		    zdev->pr_isolation_addr);
+		    domain->pr_isolation_addr);
 		return -EFAULT;
 	}
 
 	/* Freeze PR ISOLATION IP for bitstream download */
-	iowrite32(zdev->pr_isolation_freeze, map);
+	iowrite32(domain->pr_isolation_freeze, map);
 	err = zocl_fpga_mgr_load(zdev, buffer, length, FPGA_MGR_PARTIAL_RECONFIG);
 	/* Unfreeze PR ISOLATION IP */
-	iowrite32(zdev->pr_isolation_unfreeze, map);
+	iowrite32(domain->pr_isolation_unfreeze, map);
 
 	iounmap(map);
 	return err;
 }
 
+/*
+ * Load the bitstream. For PR platform load the partial bitstream and for FLAT
+ * platform load the full bitstream.
+ *
+ * @param       zdev:   zocl device structure
+ * @param       buffer: xclbin data buffer pointer
+ * @param       length: xclbin data buffer size
+ * @param       domain: Target domain structure
+ *
+ * @return      0 on success, Error code on failure.
+ */
 static int
-zocl_load_bitstream(struct drm_zocl_dev *zdev, char *buffer, int length)
+zocl_load_bitstream(struct drm_zocl_dev *zdev, char *buffer, int length,
+		   struct drm_zocl_domain *domain)
 {
 	struct XHwIcap_Bit_Header bit_header;
```

Comment:
Structure not initialized.
```suggestion
	struct XHwIcap_Bit_Header bit_header = {0};
```